### PR TITLE
Added section about preview-bundle

### PR DESCRIPTION
**Pull request by wachterjohannes**
| Q | A
| --- | ---
| Fixed tickets | none
| License | MIT

#### What's in this PR?

This PR add documentation about preview which was introduced by https://github.com/sulu/sulu/pull/2332

### CODE REVIEW

File: bundles/preview/index.rst [hunk 1]
```diff
@@ -0,0 +1,117 @@
+PreviewBundle
+=============
+
+The PreviewBundle provides the feature of preview for custom-entities.
+It is build ontop the route-bundle and can only be introduced for
```
Comment (danrot) [inline bundles/preview/index.rst]: `on top of the route-bundle`

File: bundles/preview/index.rst [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+entities which has an `RouteDefaultsProvider`.
+
+This Provider will be used to determine which controller should be
+used to render the html of the entity. This html should contain the
```
Comment (danrot) [inline bundles/preview/index.rst]: `HTML`

File: bundles/preview/index.rst [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+definition `property="..."` which will be used to find the place
+where the content takes place. See the documentation for that in
+:doc:`/cookbook/live-preview`.
+
+Example
+-------
+
+This example is a simple (and not complete) example of a news-entity form.
```
Comment (danrot) [inline bundles/preview/index.rst]: `This is a simple (and not complete) example...`

File: bundles/preview/index.rst [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+                    leftSpace: true,
+                    rightSpace: true
+                }
+            },
+
+            initialize: function() {
+                this.render();
+
+                this.sandbox.on('sulu.toolbar.save', this.save.bind(this));
+                this.options.preview.bindDomEvents(this.$el);
```
Comment (danrot) [inline bundles/preview/index.rst]: Is the preview really passed as an option?

File: bundles/route/index.rst [hunk 5]
```diff
@@ -0,0 +1,132 @@
+RouteBundle
+===========
+
+The RouteBundle provides the feature of managing website-routes for
+custom-entities. This routes will be generated by calling the
```
Comment (danrot) [inline bundles/route/index.rst]: `These routes`

File: bundles/route/index.rst [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+        }
+
+        public function supports($entityClass)
+        {
+            return $entityClass === Recipe::class;
+        }
+    }
+
+Register this class as a service with the tag `<tag name="sulu_route.defaults_provider"/>`
+and call on creating a new entity the `save` method of the service
```
Comment (danrot) [inline bundles/route/index.rst]: Don't really know what you want to say with that sentence.

**Comment by danrot** [comment]
Looks like this PR also includes https://github.com/sulu/sulu-docs/pull/245/files

File: bundles/preview/index.rst (hunk 4 above)
Comment (wachterjohannes) [inline bundles/preview/index.rst]: no only in the example i have copied from. i will fix it

**Comment by wachterjohannes** [comment]
@danrot yes i say that we merge this and close the other one.

**Comment by wachterjohannes** [comment]
@danrot i have fixed all the issue and make the last sentence of the route-bundle clearer. additionally i have added a not which mentions the `sulu:route:update` command

File: bundles/route/index.rst [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+Register this class as a service with the tag `<tag name="sulu_route.defaults_provider"/>`.
+
+After that the entity is ready to get a route. To create a route for the new entities
+simple call the `save` method of the service `sulu_route.manager.route_manager`.
+
+.. note::
+
+    To update already existing entities you can run the command
+    `console sulu:route:update AppBundle\Entity\Recipe` which updates or create the route
+    (if no exists) for all the entities of this type.
```
Comment (danrot) [inline bundles/route/index.rst]: remove `(if not exists)`, should be clear with `updates or creates`

File: bundles/route/index.rst [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+
+Register this class as a service with the tag `<tag name="sulu_route.defaults_provider"/>`.
+
+After that the entity is ready to get a route. To create a route for the new entities
+simple call the `save` method of the service `sulu_route.manager.route_manager`.
+
+.. note::
+
+    To update already existing entities you can run the command
+    `console sulu:route:update AppBundle\Entity\Recipe` which updates or create the route
```
Comment (danrot) [inline bundles/route/index.rst]: `app/console` (I know about sulu-minimal, but only `console` won't work anywhere)
`updates or creates`

**Comment by wachterjohannes** [comment]
@danrot fixed

**Pull request merged**